zamba/models/config.py

```diff
@@ -481,78 +481,105 @@ def validate_filepaths_and_labels(cls, values):
-                )
-
-                expected_splits = [k for k, v in values["split_proportions"].items() if v > 0]
-                random.seed(SPLIT_SEED)
-
-                # check we have at least as many videos per species as we have splits
-                # labels are OHE at this point
-                num_videos_per_species = labels.filter(regex="species_").sum().to_dict()
-                too_few = {
-                    k.split("species_", 1)[1]: v
-                    for k, v in num_videos_per_species.items()
-                    if v < len(expected_splits)
-                }
-
-                if len(too_few) > 0:
-                    raise ValueError(
-                        f"Not all species have enough videos to allocate into the following splits: {', '.join(expected_splits)}. A minimum of {len(expected_splits)} videos per label is required. Found the following counts: {too_few}. Either remove these labels or add more videos."
-                    )
-
-                for c in labels.filter(regex="species_").columns:
-                    species_df = labels[labels[c] > 0]
-
-                    # within each species, seed splits by putting one video in each set and then allocate videos based on split proportions
-                    labels.loc[species_df.index, "split"] = expected_splits + random.choices(
-                        list(values["split_proportions"].keys()),
-                        weights=list(values["split_proportions"].values()),
-                        k=len(species_df) - len(expected_splits),
-                    )
-
-                logger.info(f"{labels.split.value_counts()}")
-                logger.info(
-                    f"Writing out split information to {values['save_dir'] / 'splits.csv'}."
-                )
+        # if there are only two species columns and every video belongs to one of them,
+        # drop the second species column so the problem is treated as a binary classification
+        species_cols = labels.filter(regex="species_").columns
+        sums = labels[species_cols].sum(axis=1)
 
-                # create the directory to save if we need to
-                values["save_dir"].mkdir(parents=True, exist_ok=True)
-
-                labels.reset_index()[["filepath", "split"]].drop_duplicates().to_csv(
-                    values["save_dir"] / "splits.csv", index=False
-                )
+        if len(species_cols) == 2 and (sums == 1).all():
+            logger.warning(
+                f"Binary case detected so only one species column will be kept. Output will be the binary case of {species_cols[0]}."
+            )
+            labels = labels.drop(columns=species_cols[1])
 
         # filepath becomes column instead of index
         values["labels"] = labels.reset_index()
         return values
 
 
+def make_random_split(labels, values):
```

Comment:
not sure i'd call this `random` since it does do a site aware split if possible, otherwise it allocates videos entirely randomly. `make_split`, while more general, may be less misleading

Reply:
ok